### PR TITLE
Netcore: Further model unit test migrations into new project and builder pattern (2)

### PR DESCRIPTION
**Pull request by AndyButland**
In this PR I've migrated some further model unit tests into the new project and builder pattern:

- PropertyGroup
- PropertyType
- Property

I've also tweaked the setup where we have some builders that might be used in different contexts - whether standalone, created as a child of another builder and, in particular, when created as a child of more than one builder (so using generics).  I've extended this using marker interface constraints (see `PropertyGroupBuilder` and `PropertyTypeBuilder`.

### CODE REVIEW

File: src/Umbraco.Tests.Common/Builders/PropertyBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+using System;
+using Umbraco.Core.Models;
+using Umbraco.Tests.Common.Builders.Interfaces;
+
+namespace Umbraco.Tests.Common.Builders
+{
+    public class PropertyBuilder
+        : BuilderBase<Property>,
```
Comment (bergmania) [inline src/Umbraco.Tests.Common/Builders/PropertyBuilder.cs:8]: Any reason, this cannot be a `IProperty`

File: src/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/Models/PropertyTests.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                Assert.AreEqual(propertyInfo.GetValue(clone, null), propertyInfo.GetValue(property, null));
+            }
+        }
+
+        private Property BuildProperty()
+        {
+            return _builder
+                .WithId(4)
+                .WithKey(Guid.NewGuid())
+                .WithCreateDate(DateTime.Now)
```
Comment (bergmania) [inline src/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/Models/PropertyTests.cs:46]: We can use defalts for some of these.. `Key`, Dates, ...

File: src/Umbraco.Tests.Common/Builders/PropertyBuilder.cs (hunk 1 above)
Comment (AndyButland) [reply src/Umbraco.Tests.Common/Builders/PropertyBuilder.cs]: No, seems to be fine with `IProperty`, have updated.

File: src/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/Models/PropertyTests.cs (hunk 2 above)
Comment (AndyButland) [reply src/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/Models/PropertyTests.cs]: Agreed. have removed.

**Comment by AndyButland** [comment]
Have made the two suggested amends.  I agree on the extension method for a complex object, but suggest will save that for another PR?

Getting a "Packages failed to restore" though now I've merged the latest `netcore/dev` in to resolve conflicts...

**Comment by AndyButland** [comment]
Have hopefully fixed that... merge issue.

**Comment by bergmania** [comment]
Yes, a separate PR is fine for a good way to create a complex object.

**Pull request merged**